packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -485,7 +485,7 @@ export const useGridFilter = (
   /**
    * EFFECTS
    */
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
```

Comment:
You would need to use `import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/utils';` in order for SSR to work